### PR TITLE
Recognize \r as newlines from flutter

### PR DESCRIPTION
**Pull request by helin24**
Verify that opening workbench, opening project(s), quitting, then re-opening workbench consistently results in re-opened projects having device options.

Fixes https://github.com/flutter/flutter-intellij/issues/3892

### CODE REVIEW

File: src/io/flutter/utils/StdoutJsonParser.java [hunk 1]
```diff
@@ -32,11 +33,25 @@ public void appendOutput(String string) {
...
+          eatNextEol = true;
+          isPotentialWindowsReturn = true;
           continue;
         }
       }
 
+      if (isPotentialWindowsReturn) {
+        isPotentialWindowsReturn = false;
+
+        if (c != '\n') {
```
Comment (jacob314) [inline src/io/flutter/utils/StdoutJsonParser.java:50]: a little odd that we don't flush when the `c` is a `'\n'`
is this really
`assert(c!= '\n');`
?

File: testSrc/unit/io/flutter/utils/StdoutJsonParserTest.java [hunk 2]
```diff
@@ -49,6 +51,21 @@ public void appendsWithoutLineBreaks() {
     );
   }
 
+  @Test
+  public void jsonWithLineBreaks() {
+    final StdoutJsonParser parser = new StdoutJsonParser();
+    parser.appendOutput("[{'foo':'bar'}]\ntest\n");
```
Comment (jacob314) [inline testSrc/unit/io/flutter/utils/StdoutJsonParserTest.java:57]: somewhat minor: do we have to worry about `\r` or `\n` in the middle of a json block? I don't know off the top of my head how the dart json encoder deals with them.

Comment (helin24) [reply testSrc/unit/io/flutter/utils/StdoutJsonParserTest.java]: It seems this is dealt with fine, I tried putting \r, \n and \r\n in the middle of flutter tools output and it's recognized correctly.

**Comment by DanTup** [comment]
@helin24 thanks for the ping! Dart-Code did have a similar bug (if the Flutter startup lock was held by another process, it'd fail to read the `daemon.connected` event, and therefore never enable device polling).

**Pull request merged**